cloudevents-sdk-rdkafka/src/kafka_producer_record.rs

```diff
@@ -0,0 +1,104 @@
+use super::headers;
+use super::headers::{CLOUDEVENTS_JSON_HEADER, SPEC_VERSION_HEADER};
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, MessageAttributeValue, Result, StructuredSerializer,
+};
+use cloudevents::Event;
+use rdkafka::message::{OwnedHeaders, ToBytes};
+use rdkafka::producer::FutureRecord;
+
+/// Wrapper for [`RequestBuilder`] that implements [`StructuredSerializer`] & [`BinarySerializer`] traits
+#[derive(Debug)]
+pub struct ProducerRecordSerializer {
+    pub(crate) payload: Option<Vec<u8>>,
+    pub(crate) headers: OwnedHeaders,
+}
+
+impl ProducerRecordSerializer {
+    pub fn new() -> ProducerRecordSerializer {
+        ProducerRecordSerializer {
+            payload: None,
+            headers: OwnedHeaders::new(),
+        }
+    }
+}
+
+impl BinarySerializer<ProducerRecordSerializer> for ProducerRecordSerializer {
+    fn set_spec_version(mut self, spec_version: SpecVersion) -> Result<Self> {
+        self.headers = self.headers.add(SPEC_VERSION_HEADER, spec_version.as_str());
+
+        Ok(self)
+    }
+
+    fn set_attribute(mut self, name: &str, value: MessageAttributeValue) -> Result<Self> {
+        self.headers = self.headers.add(
+            &headers::ATTRIBUTES_TO_HEADERS.get(name).unwrap().clone()[..],
```

Comment:
Why this unwrap? what if the user provides an invalid `name`?

Reply:
So what should i return if the name is wrong?

Reply:
i suppose it should return some err? Does `message::Error` already have a variant like "invalid attribute name"?